Formula/zig.rb

```diff
@@ -45,6 +46,9 @@ def install
 
     (testpath/"hello.c").write <<~EOS
       #include <stdio.h>
+      #ifdef __APPLE__
+      #   include <TargetConditionals.h>
+      #endif
```

Comment:
Shouldn't think be above `#include <stdio.h>`?

Reply:
This test is here to detect a problem. A problem has been detected. This workaround is effectively the same thing as deleting the test. Let's try to figure out the problem instead of hiding it.

Reply:
Yep, was just poking at it to see if it helped. It didn't (likely because this should've been the first `#include`), but removing `CPATH` from the environment seemed to help.

Reply:
Thanks for letting me know. I'll look into making zig scrub CPATH from the environment before invoking clang.